src/Space.cpp

```diff
@@ -944,12 +965,7 @@ static void CollideWithTerrain(Body *body)
 	double terrHeight = terrain->GetTerrainHeight(body->GetPosition().Normalized());
 	if (altitude >= terrHeight) return;
 
-	CollisionContact c;
-	c.pos = body->GetPosition();
-	c.normal = c.pos.Normalized();
-	c.depth = terrHeight - altitude;
-	c.userData1 = static_cast<void *>(body);
-	c.userData2 = static_cast<void *>(f->GetBody());
+	CollisionContact c(body->GetPosition(), c.pos.Normalized(), terrHeight - altitude, timeStep, static_cast<void *>(body), static_cast<void *>(f->GetBody()));
```

Comment:
```suggestion
	CollisionContact c(body->GetPosition(), body->GetPosition().Normalized(), terrHeight - altitude, timeStep, static_cast<void *>(body), static_cast<void *>(f->GetBody()));
```
Don't use an uninitialized member in the object's own initialization please. I know this one's a simple copy-paste error, but it's still not good.